### PR TITLE
Immunization target-disease support

### PR DESCRIPTION
**Pull request by lalitha94**
Description
----
Added support for Immunization search by target-disease.
Exposing Immunization.protocolApplied field

PR Checklist
----
- [X] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

Screen Shot 2021-08-11 at 8 58 12 AM
Screen Shot 2021-08-11 at 8 58 29 AM
Screen Shot 2021-08-11 at 8 58 43 AM

### CODE REVIEW

**Review by joeratt — APPROVED** [review]

**Review by fpivonka — APPROVED** [review]

File: lib/resources/r4/immunization.yaml [hunk 1]
```diff
@@ -414,3 +414,23 @@ fields:
...
+  action: terminology
+  binding:
+    description: The vaccine preventable disease the dose is being administered against.
+    terminology:
+    - display: CVX
+      system: http://hl7.org/fhir/sid/cvx
+      info_link: http://hl7.org/fhir/r4/cvx.html
+    - display: SNOMED CT
+      system: http://snomed.info/sct
+      info_link: http://hl7.org/fhir/r4/snomedct.html
```
Comment (Mohammad-Isyroqi-Fathan) [inline lib/resources/r4/immunization.yaml:432]: Do we actually have SNOMED mapping for this?

Comment (lalitha94) [reply lib/resources/r4/immunization.yaml:432]: Thanks for pointing it out. I think currently we do not have mappings but we do want to support it. @fpivonka can we leave it or should we add it once the mappings are completed?

Comment (fpivonka) [reply lib/resources/r4/immunization.yaml:432]: @lalitha94 @Mohammad-Isyroqi-Fathan I think we can leave it off for now. We do support the scope for SNOMED CT but we have not mapped clients that way.

**Comment by ashu-bdvt** [comment]
Post deployment screenshots:

Screenshot 2021-09-09 at 8 02 30 PM
Screenshot 2021-09-09 at 8 00 49 PM
Screenshot 2021-09-09 at 8 00 24 PM

**Pull request merged**